homeassistant/components/caldav/calendar.py

```diff
@@ -193,27 +193,49 @@ def __init__(self, calendar, include_all_day, search):
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
         """Get the latest data."""
+        start_of_today = dt.start_of_local_day()
+        start_of_tomorrow = dt.start_of_local_day() + timedelta(days=1)
+
         # We have to retrieve the results for the whole day as the server
         # won't return events that have already started
-        results = self.calendar.date_search(
-            dt.start_of_local_day(), dt.start_of_local_day() + timedelta(days=1)
-        )
+        results = self.calendar.date_search(start_of_today, start_of_tomorrow)
+
+        # Create new vevents for each recurrence of an event that happens today.
+        # For recurring events, some servers return the original event with recurrence rules
```

Comment:
> some servers return the original event with recurrence rules

This is exactly how RFC5545 works. Which calendar does not do this? 🤔

Reply:
Well until this PR I was using Nextcloud and my recurring events were working properly. Then I switched to SOGo and they stopped working. So if our code was perfectly compliant with the RFC, then one of those servers must not be. 🤷‍♂

Reply:
Well are both returning you RFC5545 formatted events (ical) or are some given you "normalized" events where they pre-process recurring events for you?

Calendaring is hard, I am not surprised if certain servers mess it up.

Reply:
[RFC4971](https://tools.ietf.org/html/rfc4791#section-7.8.3) talks about "expand", which the python `caldav` package does send to the server. It doesn't appear that the SOGo CalDAV implementaiton honors it.